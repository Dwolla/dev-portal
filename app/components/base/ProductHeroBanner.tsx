import styled from "@emotion/styled";
import { Button } from "@mui/material";
import React from "react";
import { POPPINS, ROBOTO } from "../typography";
import { BALANCE_HERO_GRADIENT, CONNECT_HERO_GRADIENT } from "../colors";
import { breakDown } from "../breakpoints";
import { BALANCE_ICON_SHADOW, CONNECT_ICON_SHADOW } from "../shadowDepths";

// Define type for the Product variant options
type ProductVariant = "Balance" | "Connect";

// Styled Banner component
const StyledBanner = styled.div<{ variant: ProductVariant }>`
  width: 100%;
  padding: 3em;
  display: flex;
  justify-content: space-around;
  align-items: center;
  position: relative;
  border-radius: 16px;
  background: ${(props) => {
    if (props.variant === "Balance") {
      return BALANCE_HERO_GRADIENT;
    }
    if (props.variant === "Connect") {
      return CONNECT_HERO_GRADIENT;
    }
    return "transparent";
  }}; // Set background based on variant
`;

const BannerContentWrap = styled.div`
  max-width: 525px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;

  > *:last-child {
    margin-bottom: 0;
  }
`;

const StyledTopic = styled.div`
  color: white;
  font-family: ${POPPINS};
  font-style: normal;
  font-weight: 500;
  font-size: 40px;
  line-height: 123.5%;
  margin-bottom: 32px;
  letter-spacing: 0.25px;

  @media (${breakDown("sm")}) {
    font-size: 30px;
  }

  @media (${breakDown("xs")}) {
    font-size: 22px;
  }
`;

const StyledDescription = styled.div`
  font-family: ${ROBOTO};
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 175%;
  letter-spacing: 0.15px;
  margin-bottom: 24px;
  color: white;

  @media (${breakDown("sm")}) {
    font-size: 12px;
  }
`;

const ButtonsWrap = styled.div`
  display: block;

  a {
    margin: 10px;
  }
`;

const HeroGraphicStyle = styled.div<{ variant: ProductVariant }>`
  svg {
    filter: drop-shadow(
      ${(props) => {
        if (props.variant === "Balance") {
          return BALANCE_ICON_SHADOW;
        }
        if (props.variant === "Connect") {
          return CONNECT_ICON_SHADOW;
        }
        return "none";
      }}
    ); // Set SVG filter based on variant
  }
`;

type LinkProp = { text: string; href: string };

type Props = {
  variant: ProductVariant;
  HeroGraphic?: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
  topic: string;
  description: string;
  links?: Array<LinkProp>;
};

function ProductHeroBanner({
  variant,
  HeroGraphic,
  topic,
  description,
  links,
}: Props) {
  return (
    <StyledBanner variant={variant}>
      <BannerContentWrap>
        <StyledTopic>{topic}</StyledTopic>
        <StyledDescription>{description}</StyledDescription>
        {links && (
          <ButtonsWrap>
            {links.map((link) => (
              <Button
                key={link.text}
                href={link.href}
                variant="outlined"
                color="white"
              >
                {link.text}
              </Button>
            ))}
          </ButtonsWrap>
        )}
      </BannerContentWrap>
      <HeroGraphicStyle variant={variant}>
        <HeroGraphic height="10em" width="10em" />
      </HeroGraphicStyle>
    </StyledBanner>
  );
}

export default ProductHeroBanner;
