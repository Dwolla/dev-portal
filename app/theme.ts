import { createTheme } from "@mui/material";
import { red } from "@mui/material/colors";
import {
  ORANGE_PRIMARY,
  PURPLE_075,
  PURPLE_087,
  PURPLE_PRIMARY,
  WHITE_PRIMARY,
} from "./components/colors";
import { POPPINS, ROBOTO } from "./components/typography";

export default createTheme({
  palette: {
    primary: {
      main: PURPLE_PRIMARY,
    },
    secondary: {
      main: ORANGE_PRIMARY,
    },
    error: {
      main: red.A400,
    },
    white: {
      main: WHITE_PRIMARY,
    },
  },
  typography: {
    button: {
      fontFamily: ROBOTO,
      fontWeight: 500,
      fontSize: "15px",
      lineHeight: "26px",
      letterSpacing: "0.46px",
      textTransform: "capitalize",
    },
    h2: {
      fontFamily: POPPINS,
      fontStyle: "normal",
      fontWeight: 600,
      fontSize: "32px",
      lineHeight: "123.5%",
      letterSpacing: "0.25px",
      color: PURPLE_087,
    },
    body1: {
      fontFamily: ROBOTO,
      fontStyle: "normal",
      fontWeight: 400,
      fontSize: "16px",
      lineHeight: "175%",
      letterSpacing: "0.15px",
      color: PURPLE_075,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 992,
      xl: 1200,
    },
  },
});
