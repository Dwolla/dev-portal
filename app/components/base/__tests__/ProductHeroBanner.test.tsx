import renderer from "react-test-renderer";
import { ThemeProvider } from "@mui/material";
import ProductHeroBanner from "../ProductHeroBanner";
import { ReactComponent as HeroGraphic } from "../../../../assets/images/product-icons-and-heroes/dwolla-balance-icon-48x48.svg";
import theme from "../../../theme";

test("FooterCTA", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ProductHeroBanner
          HeroGraphic={HeroGraphic}
          topic="Dwolla Balance"
          description="Use Dwollaâ€™s existing bank partnerships to connect your software to the banking infrastructure."
          links={[
            {
              text: "API reference",
              href: "/api-reference",
            },
            {
              text: "Code Samples",
              href: "/code-samples",
            },
            {
              text: "SDKs",
              href: "/sdks-tools",
            },
          ]}
        />
      </ThemeProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
