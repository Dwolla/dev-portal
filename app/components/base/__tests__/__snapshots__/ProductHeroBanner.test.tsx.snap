// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterCTA 1`] = `
<div
  className="css-88pzot-StyledBanner esy61rv5"
>
  <div
    className="css-obbnl1-BannerContentWrap esy61rv4"
  >
    <div
      className="css-160rdza-StyledTopic esy61rv3"
    >
      Dwolla Balance
    </div>
    <div
      className="css-1wh94x8-StyledDescription esy61rv2"
    >
      Use Dwollaâ€™s existing bank partnerships to connect your software to the banking infrastructure.
    </div>
    <div
      className="css-lepm0g-ButtonsWrap esy61rv1"
    >
      <a
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedWhite MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedWhite MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1l1s02k-MuiButtonBase-root-MuiButton-root"
        href="/api-reference"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        API reference
      </a>
      <a
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedWhite MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedWhite MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1l1s02k-MuiButtonBase-root-MuiButton-root"
        href="/code-samples"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        Code Samples
      </a>
      <a
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedWhite MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedWhite MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1l1s02k-MuiButtonBase-root-MuiButton-root"
        href="/sdks-tools"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        SDKs
      </a>
    </div>
  </div>
  <div
    className="css-zlmowv-HeroGraphicStyle esy61rv0"
  >
    <svg
      height="10em"
      width="10em"
    />
  </div>
</div>
`;
